tests/tt_metal/tt_metal/test_create_kernel_from_string.cpp

```diff
@@ -0,0 +1,159 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+
+#include <unordered_set>
+
+#include "core_coord.h"
+#include "detail/tt_metal.hpp"
+#include "host_api.hpp"
+#include "impl/device/device.hpp"
+#include "impl/kernels/data_types.hpp"
+#include "impl/kernels/kernel_types.hpp"
+#include "impl/program/program.hpp"
+#include "tt_cluster_descriptor_types.h"
+
+using namespace tt;
+using namespace tt::tt_metal;
+
+class CompileProgramWithKernelCreatedFromString : public ::testing::Test {
+   protected:
+    void SetUp() override {
+        vector<chip_id_t> device_ids;
+        const uint32_t num_devices = GetNumAvailableDevices();
+        for (chip_id_t i = 0; i < num_devices; i++) {
+            device_ids.push_back(i);
+        }
+        this->devices_ = detail::CreateDevices(device_ids);
+    }
+
+    void TearDown() override { detail::CloseDevices(this->devices_); }
+
+    std::map<chip_id_t, Device *> devices_;
+};
+
+TEST_F(CompileProgramWithKernelCreatedFromString, DataMovementKernel) {
```

Comment:
I think it would be useful to add some tests that run programs on device with kernels created from string or extend these to run

Reply:
I've modified the tests to call `CommonFixture::RunProgram()` to run after the program has been compiled.